### PR TITLE
Adding types for vrld/hump

### PR DESCRIPTION
**Pull request by profan**
So I came to start using teal as I wanted to write games in Lua again but remembered what happened as my old game project grew and so here we are, some crude bindings for https://github.com/vrld/hump 👀 

May come to revise these later if I find some bugs in them as I'll be using them myself at least

### CODE REVIEW

File: types/hump/camera.d.tl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+local record camera
+
+    type DrawFunction = function()
+    type SmoothingFunction = function(x: number, y: number, s: number): number, number
+
+    new: function(x: number, y: number, zoom: number, rot: number): camera
```
Comment (pdesaulniers) [inline types/hump/camera.d.tl:6]: Apparently, it is also possible to create a new camera by calling the module as a function:

```lua
    metamethod __call: function(self: camera, x: number, y: number, zoom: number, rot: number): camera
```

**Review by pdesaulniers — COMMENTED** [review]
Great work! I gave a quick look at `camera.d.tl` and left a couple of comments :)

File: types/hump/camera.d.tl (hunk 1 above)
Comment (profan) [reply types/hump/camera.d.tl:6]: ah right i missed that part for the camera, reckon it worked the same as the vectors

File: types/hump/camera.d.tl [hunk 2]
```diff
@@ -0,0 +1,39 @@
+local record camera
+
+    type DrawFunction = function()
+    type SmoothingFunction = function(x: number, y: number, s: number): number, number
```
[the remark this answers is not in the document]
Comment (profan) [reply types/hump/camera.d.tl]: added

File: types/hump/camera.d.tl [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    attach: function(self: camera)
+    detach: function(self: camera)
+
+    draw: function(self: camera, func: DrawFunction)
+    worldCoords: function(self: camera, x: number, y: number): number, number
+    cameraCoords: function(self: camera, x: number, y: number): number, number
+    mousePosition: function(self: camera): number, number
+
+    -- no extra args to lockX, lockY atm, TODO
```
[the remark this answers is not in the document]
Comment (profan) [reply types/hump/camera.d.tl:32]: added

File: types/hump/camera.d.tl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    lockX: function(self: camera, x: number, smoother: SmoothingFunction)
+    lockY: function(self: camera, y: number, smoother: SmoothingFunction)
+    lockPosition: function(self: camera, x: number, y: number, smoother: SmoothingFunction)
+    lockWindow: function(self: camera, x: number, y: number, x_min: number, x_max: number, y_min: number, y_max: number, smoother: SmoothingFunction)
+
+    smoother: SmoothingFunction
+    scale: number
+    x: number
+    y: number
+
```
[the remark this answers is not in the document]
Comment (profan) [reply types/hump/camera.d.tl:45]: added :)

File: types/hump/camera.d.tl (hunk 1 above)
Comment (profan) [reply types/hump/camera.d.tl:12]: added

**Comment by profan** [comment]
Thanks for the merge and review! If I find any holes in this stuff I'll come back with PRs to plug them :)

**Pull request merged**